### PR TITLE
[UI] Colorful Gates

### PR DESCRIPTION
**Pull request by BartoszMichalak13**

### CODE REVIEW

**Review by kociap — CHANGES_REQUESTED** [review]
lots of pointless comments or commented code.

File: src/model/gate.hpp [hunk 1]
```diff
@@ -38,6 +39,32 @@ namespace nebula {
...
+      number = number - math::floor(number);
+    }
+    color = number;
+  }
+
+  inline void gate_colors_init() {
+    u8 number_of_gate_types = static_cast<int>(Gate_Kind::e_count);
+    i64 density = math::ceil(math::cbrt(number_of_gate_types));
+    for(int i = 0; i < number_of_gate_types; ++i) {
+      f32 red,green,blue,number1,number2,number3 = 0;
```
Comment (kociap) [inline src/model/gate.hpp]: what are number1, number2, number3?

File: src/model/gate.cpp [hunk 2]
```diff
@@ -63,11 +63,15 @@ namespace nebula {
...
-    math::Vec3 const color = gate.evaluation.value ? green : red;
+    // TODO: change color scheme for evaluation phase
+
+    // if (run_evaluation) {
+      // math::Vec3 const green{0.498f, 1.0f, 0.0f};
+      // math::Vec3 const red{1.0f, 0.0f, 0.2235f};
+      // math::Vec3 const color = gate.evaluation.value ? green : red;
+    // } else {
+      math::Vec3 const color = gate_colors[static_cast<int>(gate.kind)];
+    // }
```
Comment (kociap) [inline src/model/gate.cpp]: this breaks displaying the value of the gate

**Review by kociap — CHANGES_REQUESTED** [review]

File: src/model/gate.cpp (hunk 2 above)
Comment (BartoszMichalak13) [reply src/model/gate.cpp]: Repaired

File: src/model/gate.hpp (hunk 1 above)
Comment (BartoszMichalak13) [reply src/model/gate.hpp]: Removed, area cleared

**Pull request closed without merging**